### PR TITLE
refactor(app): setupLabwareList labwareId logic refactor

### PR DESCRIPTION
**Pull request by jerader**
closes RAUT-270

# Overview

the `SetupLabwareList` component had some logic that needed to be refactored to match the labware key/value structure

# Changelog

- change logic in `getAllLabwareAndTiprackIdsInOrder` and fix test

# Review requests

- smoke test to make sure everything works as expected and there is no whitescreen. Smoke test: protocol setup, run protocol, and LPC

# Risk assessment

low

### CODE REVIEW

File: app/src/organisms/Devices/hooks/useStoredProtocolAnalysis.ts [hunk 1]
```diff
@@ -73,6 +73,7 @@ export function useStoredProtocolAnalysis(
 
   // @ts-expect-error
   return storedProtocolAnalysis != null && 'modules' in storedProtocolAnalysis
-    ? schemaV6Adapter(storedProtocolAnalysis)
+    ? // @ts-expect-error
+      schemaV6Adapter(storedProtocolAnalysis)
```
Comment (jerader) [inline app/src/organisms/Devices/hooks/useStoredProtocolAnalysis.ts:77]: Not sure why I needed to add this `ts-expect-error`, maybe it was accidentally deleted from when you rebased @smb2268?

File: app/src/organisms/Devices/ProtocolRun/SetupLabware/utils.ts [hunk 2]
```diff
@@ -1,54 +1,36 @@
...
+  labware: LoadedLabware[],
   labwareDefinitions: Record<string, LabwareDefinition2>,
   commands: RunTimeCommand[]
 ): string[] => {
-  const unorderedLabware = reduce<typeof labware, LabwareToOrder[]>(
-    labware,
-    (unorderedLabware, currentLabware, labwareId) => {
-      const labwareDef = labwareDefinitions[currentLabware.definitionId]
-      const labwareLocation = getLabwareLocation(labwareId, commands)
+  const unorderedLabware = labware.reduce<LabwareToOrder[]>(
```
Comment (smb2268) [inline app/src/organisms/Devices/ProtocolRun/SetupLabware/utils.ts]: we should probably rename this `orderedLabware` here and on line 19

File: app/src/organisms/Devices/hooks/useStoredProtocolAnalysis.ts (hunk 1 above)
Comment (smb2268) [reply app/src/organisms/Devices/hooks/useStoredProtocolAnalysis.ts:77]: yep that must be what happened

**Review by smb2268 — APPROVED** [review]
One small naming comment but looks great otherwise!

**Pull request merged**